### DISCUSSION TITLE
No EXIF data but XMP sidecar works

### DISCUSSION THREAD

**Discussion by Qhilm** [RhetTbull/osxphotos#1673] [Q&A]
It appears as if exiftool is failing, but there is no error. The resulting file has no EXIF. Adding `--sidecar XMP` produces XMP sidecars with the correct data (I'm mostly interested in the date).

If I open the file on macOS and press Command-I, under info, there's strange `{general}` section, which is empty):

<img width="322" alt="image" src="https://github.com/user-attachments/assets/8dd7b10b-5e7c-4d47-8e8d-cb884fb971e3">

The originals are scans, meaning they do not have any EXIF data, but my understanding is that exiftool is smart enough to create the entire exif block if it cannot find it.

How can I troubleshoot why exiftool is failing here?

The command:

`./venv/bin/osxphotos export -VV --library "/Volumes/microSDXC/Pictures/Library1.photoslibrary" --album "0606 : Excursion" --skip-original-if-edited --exiftool --ignore-date-modified --keyword-template "{folder_album(>)}" --sidecar XMP --directory '{folder_album}' "/Volumes/microSDXC/Pictures/Library1 – export"`

osxphotos 0.68.5 (installed via pip in a venv)
exiftool 12.76
macOS 14.6.1

**Comment by RhetTbull** [comment]
If exiftool generated an error or a warning then OSXPhotos should have displayed this. (see example below). If you're able to share one of the images with me at osxphotos.py@gmail.com I'll take a look.  You could also try running exiftool manually on one of the original images with the `-verbose` flag:

`exiftool -verbose -IPTC:Keywords=Test test.jpg`

```
❌️  exiftool error for file /Users/rhet/Desktop/export/IMG_5339 (3).HEIC: Error: Not a valid HEIC (looks more like a JPEG) -
/var/folders/sb/8031jv2x2qs3pj3prpk225g40000gn/T/osxphotos_export_szh8c0s8/4DE8871A-DB69-43E1-BFC8-ABFCB4BD662B_exiftool.heic
❌️  Error exporting photo (4DE8871A-DB69-43E1-BFC8-ABFCB4BD662B: IMG_5339.HEIC) as /Users/rhet/Desktop/export/IMG_5339 (3).HEIC: Error: Not a valid HEIC (looks more like a JPEG)
- /var/folders/sb/8031jv2x2qs3pj3prpk225g40000gn/T/osxphotos_export_szh8c0s8/4DE8871A-DB69-43E1-BFC8-ABFCB4BD662B_exiftool.heic
 ```

**Comment by Qhilm** [comment]
As usual, the issue was so simple I couldn't see it: the issue seems to be with Apple Preview.

Preview is unable to read exif data on on these specific files, but pixelmator can, Immich can as well, seems it's a bug in Preview for these specific files...

**Comment by Qhilm** [comment]
Interestingly though, the exiftool command results in a file which does not present the same problem, Preview is able to read the exif without issue.

```shell
$ exiftool -verbose -IPTC:Keywords=Test IMG_0027.jpg 
======== IMG_0027.jpg
Rewriting IMG_0027.jpg...
  Editing tags in: APP13 IPTC Photoshop 
JPEG APP0 (14 bytes)
Creating APP13:
  Creating Photoshop
  Creating IPTC
JPEG COM (10 bytes)
JPEG DQT (130 bytes)
JPEG DHT (416 bytes)
JPEG SOF0:
JPEG SOS
    1 image files updated
```

**Reply by Qhilm** [reply]
`exiftool -verbose '-datetimeoriginal=1974:01:01 12:00:00' IMG_0027.jpg` also works and the resulting file is readable for Apple Preview.

Something else is making it unreadable for Preview – which does not exclude at all a bug in Preview.
